pkg/exasol-rest-api/integration_test.go

```diff
@@ -0,0 +1,207 @@
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusOK, responseRecorder.Code)
+	suite.Equal("{\"status\":\"ok\",\"responseData\":{\"results\":[{\"resultType\":\"resultSet\",\"resultSet\":{\"numColumns\":2,\"numRows\":1,\"numRowsInMessage\":1,\"columns\":[{\"name\":\"X\",\"dataType\":{\"type\":\"DECIMAL\",\"precision\":18,\"scale\":0}},{\"name\":\"Y\",\"dataType\":{\"type\":\"VARCHAR\",\"size\":100,\"characterSet\":\"UTF8\"}}],\"data\":[[15],[\"test\"]]}}],\"numResults\":1}}",
+		responseRecorder.Body.String())
+}
+
+func (suite *IntegrationTestSuite) TestInsertNotAllowed() {
+	router := suite.startServer(suite.createApplicationWithDefaultProperties())
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/CREATE SCHEMA not_allowed_schema", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusOK, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
+		"{\"status\":\"error\",\"exception\":{\"text\":\"insufficient privileges for creating schema")
+}
+
+func (suite *IntegrationTestSuite) TestExasolUserWithoutCreateSessionPrivilege() {
+	username := "user_without_session"
+	password := "secret"
+	suite.createExasolUser(username, password)
+	router := suite.startServer(suite.createApplication(&exasol_rest_api.ApplicationProperties{
+		ExasolUser:                username,
+		ExasolPassword:            password,
+		ExasolPort:                suite.exasolPort,
+		ExasolWebsocketApiVersion: 2,
+	}))
+
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusBadRequest, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
+		"{\"ErrorCode\":\"EXA-REST-API-1\",\"Message\":\"[08004] Connection exception - insufficient privileges: CREATE SESSION.\"}")
+}
+
+func (suite *IntegrationTestSuite) TestExasolUserWithWrongCredentials() {
+	router := suite.startServer(suite.createApplication(&exasol_rest_api.ApplicationProperties{
+		ExasolUser:                "not_existing_user",
+		ExasolPassword:            "wrong_password",
+		ExasolPort:                suite.exasolPort,
+		ExasolWebsocketApiVersion: 2,
+	}))
+
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusBadRequest, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
+		"{\"ErrorCode\":\"EXA-REST-API-1\",\"Message\":\"[08004] Connection exception - authentication failed.\"}")
```

Comment:
Also not clear to me which authentication.
```suggestion
		"{\"ErrorCode\":\"EXA-REST-API-1\",\"Message\":\"[08004] Connection exception - Failed to authentificate at the Exasol database. Please check the configured ExasolUser ('<USER_NAME') and ExasolPassword.\"}")
```

Reply:
And this is also from the database :(

Reply:
I guess our database needs better error messages

Reply:
I agree. But we need to work for now with what we have ;-)